bugsnag-android-core/src/test/java/com/bugsnag/android/DocumentPathTest.kt

```diff
@@ -0,0 +1,191 @@
+        val value = 1
+        val expected = mapOf<String, Any>("a" to 1)
+
+        val actual = docpath.modifyDocument(document, value)
+        Assert.assertEquals(expected, actual)
+    }
+
+    @Test
+    fun testMapModify() {
+        val document = HashMap<String, Any>(mapOf("a" to 1))
+        val docpath = DocumentPath("a")
+        val value = 2
+        val expected = mapOf<String, Any>("a" to 2)
+
+        val actual = docpath.modifyDocument(document, value)
+        Assert.assertEquals(expected, actual)
+    }
+
+    @Test
+    fun testMapDelete() {
+        val document = HashMap<String, Any>(mapOf("a" to 1))
+        val docpath = DocumentPath("a")
+        val value = null
+        val expected = mapOf<String, Any>()
+
+        val actual = docpath.modifyDocument(document, value)
+        Assert.assertEquals(expected, actual)
+    }
+
+    @Test
+    fun testMapLevel2Put() {
+        val document = HashMap<String, Any>()
+        val docpath = DocumentPath("a.b")
+        val value = 1
+        val expected = mapOf("a" to mapOf("b" to 1))
+
+        val actual = docpath.modifyDocument(document, value)
+        Assert.assertEquals(expected, actual)
+    }
+
+    @Test
+    fun testMapLevel2PutEscapes() {
+        val document = HashMap<String, Any>()
+        val docpath = DocumentPath("a\\.x.b\\.y")
+        val value = 1
+        val expected = mapOf("a.x" to mapOf("b.y" to 1))
+
+        val actual = docpath.modifyDocument(document, value)
+        Assert.assertEquals(expected, actual)
+    }
+
+    @Test(expected = IllegalArgumentException::class)
+    fun testBadEscape() {
+        DocumentPath("a\\.x.b\\.y\\")
+    }
+
+    @Test
+    fun testListAppendEmpty() {
+        val document = HashMap<String, Any>()
+        val docpath = DocumentPath("a.-1")
+        val value = 1
+        val expected = mapOf<String, Any>("a" to listOf<Any>(1))
+
+        val actual = docpath.modifyDocument(document, value)
+        Assert.assertEquals(expected, actual)
+    }
+
+    @Test
+    fun testListDeleteLast() {
+        val document = HashMap<String, Any>(
+            mapOf(
+                "a" to LinkedList<Any>(listOf<Any>(1, 2, 3))
+            )
+        )
+        val docpath = DocumentPath("a.-1")
+        val value = null
+        val expected = mapOf<String, Any>("a" to listOf<Any>(1, 2))
+
+        val actual = docpath.modifyDocument(document, value)
+        Assert.assertEquals(expected, actual)
+    }
+
+    @Test
+    fun testListAppendEmptyNull() {
+        val document = HashMap<String, Any>()
+        val docpath = DocumentPath("a.-1")
+        val value = null
+        val expected = mapOf<String, Any>("a" to listOf<Any>())
+
+        val actual = docpath.modifyDocument(document, value)
+        Assert.assertEquals(expected, actual)
+    }
+
+    @Test
+    fun testListReplace() {
+        val document = HashMap<String, Any>(mapOf<String, Any>("a" to LinkedList<Any>(listOf<Any>(1, 2, 3))))
```

Comment:
Surely there's a more compact way to convey `{"a": [1, 2, 3]}`? Seems a bit overkill to need to extract this.

Reply:
Type inference and using `mutableMapOf` would allow for this to be golfed down a bit:

`val document = mutableMapOf<String, Any>("a" to LinkedList(listOf(1, 2, 3)))`

Maybe formatting the value onto separate lines would help any test cases which are more complex than this.

Reply:
`HashMap` has a convenient `hashMapOf` method, so this could be shortened a little:

```suggestion
        val document = hashMapOf<String, Any>("a" to LinkedList<Any>(listOf<Any>(1, 2, 3)))
```